src/client/Microsoft.Identity.Client.Brokers/BrokerExtension.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.PlatformsCommon.Shared;
+
+namespace Microsoft.Identity.Client.Broker
+{
+    /// <summary>
+    /// MSAL Broker Extension for WAM support
+    /// </summary>
+    public static class BrokerExtension
+    {
+        /// <summary>
+        /// Enables MSAL to use Broker flows, which are more secure than browsers. 
+        /// For details about Windows broker, see https://aka.ms/msal-net-wam
+        /// </summary>
+        /// <remarks>
+        /// No broker integration exists on Mac and Linux yet. 
+        /// Windows broker does not work on Win 8, Win Server 2016 and lower.         
+        /// If a broker does not exist or cannot be used, MSAL will fallback to a browser. Make sure browser auth is enabled (e.g. if using system browser, register the "http://localhost" redirect uri etc.)
+        /// </remarks>
+        public static PublicClientApplicationBuilder WithBroker2(this PublicClientApplicationBuilder builder, bool enableBroker = true) // TODO: naming
```

Comment:
I think "WithMSALBroker" or something similar may be better since we are now using a version of MSAL C++ underneath instead of the "web account manager". We can change it to WithBroker in MSAL 5 I guess.

thoughts? @bgavrilMS

Reply:
We already have a `WithBroker` API in MSAL. Not sure if `WithMSALBroker` is a clearer name. I was thinking: 

- we can still `WithBroker` since it is in a different namespace (this may confuse some people though)
- `WithBroker2` because this is a temporary solution. When "broker2" is at parity with "broker1" we can move it into MsalDesktop extension.

Reply:
WithBroker can be confusing, there should be a differentiator

Reply:
Based on our discussion we are naming it as `WithBrokerPreview`